### PR TITLE
Refactoring ResponseQueueReader to use Row instead of ReadRowResponse.

### PR DESCRIPTION
**Pull request by sduskis**
Also, using StreamReader interface in ResponseQueueReader.

This change will help towards #703.

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 1]
```diff
@@ -336,20 +336,6 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
...
-    int batchRequestSize;
-
-    if (isGet) {
-      // Batch request size is more performant with a value of 1 for single row gets, while a higher
-      // number is more performant for scanning
-      batchRequestSize = 1;
-      streamingBufferSize = 10;
-    } else {
-      batchRequestSize = retryOptions.getStreamingBatchSize();
-      streamingBufferSize = retryOptions.getStreamingBufferSize();
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: Are these two options deprecated now?

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ReadRowsResponseListener.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  public ReadRowsResponseListener(
+      StreamObserver<Row> observer, ClientCall<?, ReadRowsResponse> call) {
+    this.observer = observer;
+    this.call = call;
+  }
+
+  @Override
+  public void onMessage(ReadRowsResponse response) {
+    call.request(1);
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ReadRowsResponseListener.java]: Does it still make sense to request a batch at a time?

**Comment by sduskis** [comment]
I readded the batch reuests.  PTAL.

**Comment by garye** [comment]
LGTM

**Pull request merged**